content/backend/typescript-apollo/9-deployment.md

```diff
@@ -0,0 +1,692 @@
+---
+title: Deploying to Production
+pageTitle: 'Continuous Deployment Tutorial with Heroku and GitHub Actions'
+description: 'Inside .'
+question:
+  When does a workflow get triggered automatically on GitHub Actions?
+answers: ['Workflows need to be triggered manually', 'Workflows are triggered after every push', 'Workflows are triggered based on configurable events', 'Workflows are triggered when any underlying job is triggered']
+correctAnswer: 2
+---
+
+In this section of the tutorial, you will be deploying your API to production so that it can be used by you and others. Unlike the previous sections, you will not develop any new features for your API in this chapter. Instead, you will refactor the project to make it suitable for deployment on [Heroku](https://dashboard.heroku.com/). Additionally, you will automate the deployment process using [GitHub Actions](https://docs.github.com/en/actions) (this is also referred to as [continuous deployment](https://en.wikipedia.org/wiki/Continuous_deployment)). 
+
+### Prerequisites 
+
+You will need to install a few command line tools and create some accounts to follow along with this chapter. We suggest you do this now so as not to break the flow of the tutorial.
+
+- **GitHub (Account)**: Create a free account on [GitHub](https://github.com/join).
+- **GitHub (CLI)**: Install the latest stable version of the [GitHub CLI](https://cli.github.com/). Afterward, [login to your GitHub account](https://cli.github.com/manual/gh_auth_login) on the CLI. 
+- **Heroku (Account)**: Create a free account on [Heroku](https://signup.heroku.com/). 
+- **Heroku (CLI)**: Install the latest stable version of the [Heroku CLI](https://devcenter.heroku.com/articles/heroku-cli). Afterward, [login to your Heroku account](https://devcenter.heroku.com/articles/heroku-cli#get-started-with-the-heroku-cli) on the Heroku CLI. 
+- **[Docker](https://docs.docker.com/get-docker/)** and **[Docker Compose](https://docs.docker.com/compose/install/)**
+
+> **Note:** Docker and Docker Compose will be used to run an instance of the PostgreSQL database. If you prefer, you can install and run [PostgreSQL](https://www.postgresql.org/download/) natively as well. If you choose to run natively, you will need to make some slight changes to the instructions shown in this tutorial. 
```

Comment:
> Could you mention what these changes are? Will these changes also be highlighted in the tutorial?

The changes won't be highlighted. But it's basically just changing the Prisma connection string and not starting the docker container. 

These steps aren't explicitly mentioned though as it would add unnecessary complexity for anyone using the containerized database.

Reply:
> If you choose to run PostgreSQL locally 

Is locally the right word here? Because the containerized version is also running "locally".

Reply:
I'm not entirely sure. I will leave it up to you to choose the right terminology here.